app/models/miq_server/worker_management/monitor.rb

```diff
@@ -23,6 +23,9 @@ def monitor_workers
     # Sync the workers after sync'ing the child worker settings
     sync_workers
 
+    starting = MiqWorker.find_starting.find_all { |w| MiqWorkerType.worker_class_names.include?(w.class.name) }.to_a
+    sync_starting_workers!(starting)
```

Comment:
I wonder if (@kbrock like your opinion on this) we could return the list of starting workers from this method instead of passing it in.

We're not using `starting` here after `sync_starting_workers` and in `wait_for_started_workers` the first thing we do is call 
```
starting = MiqWorker.find_starting.find_all { |w| MiqWorkerType.worker_class_names.include?(w.class.name) }.to_a
sync_starting_workers!(starting)
```

So we end up with the `starting = MiqWorker.find_starting.....` duplicated in two places

Reply:
I agree with this, but fyi If we move `starting = MiqWorker.find_starting.....` to the `sync_starting_workers` method then it will still be duplicated in two places, since the method exists in `systemd.rb` and `kubernetes.rb` where the list of starting workers is needed

Reply:
This will be duplicated in 3 spots, so make a common method for this and move it into `sync_starting_workers`.

Looks like systemd version may even have another selector for non rails worker?